R/funnelplot.R

```diff
@@ -0,0 +1,774 @@
+          } else {
+            fitSummary$k <- fits[[i]]$k
+          }
+
+          return(fitSummary)
+        }))
+
+        rankCorrelationTable$setData(fitSummaries)
+
+      }
+    }
+  }
+
+  return()
+}
+
+.fpComputeFunnelDf           <- function(seSeq, mean, heterogeneity, funnelLevels) {
+  dfs <- list()
+  for (i in seq_along(funnelLevels)) {
+    tempZ <- qnorm(funnelLevels[i], lower.tail = FALSE)
+    dfs[[i]] <- data.frame(
+      x = c(rev(mean - tempZ * sqrt(heterogeneity^2 + seSeq^2)), mean + tempZ * sqrt(heterogeneity^2 + seSeq^2)),
+      y = c(rev(seSeq), seSeq),
+      p = 2*funnelLevels[i],
+      lvl = 1-2*funnelLevels[i]
+    )
+  }
+  return(dfs)
+}
+.fpAssymetryTestErrorMessage <- function(level = NULL) {
+  if (is.null(level))
+    return(gettext("The funnel plot assymetry test failed with the following error: "))
+  else
+    return(gettextf("The funnel plot assymetry test at level %1$s failed with the following error: ", level))
+}
+.dpExtractAsymmetryTest      <- function(fitTest, testType) {
+  if (testType == "metaRegression") {
+    return(data.frame(
+      k   = if (jaspBase::isTryError(fitTest)) NA else fitTest$fit$k, # nobs will be fixed in the next release
+      z   = if (jaspBase::isTryError(fitTest)) NA else fitTest$zval,
+      p   = if (jaspBase::isTryError(fitTest)) NA else fitTest$pval,
+      est = if (jaspBase::isTryError(fitTest)) NA else fitTest$est,
+      lCI = if (jaspBase::isTryError(fitTest)) NA else fitTest$ci.lb,
+      uCI = if (jaspBase::isTryError(fitTest)) NA else fitTest$ci.ub
+    ))
+  } else if (testType == "weightedRegression") {
+    return(data.frame(
+      k   = if (jaspBase::isTryError(fitTest)) NA else nobs(fitTest$fit),
+      t   = if (jaspBase::isTryError(fitTest)) NA else fitTest$zval,
+      df  = if (jaspBase::isTryError(fitTest)) NA else fitTest$dfs,
+      p   = if (jaspBase::isTryError(fitTest)) NA else fitTest$pval,
+      est = if (jaspBase::isTryError(fitTest)) NA else fitTest$est,
+      lCI = if (jaspBase::isTryError(fitTest)) NA else fitTest$ci.lb,
+      uCI = if (jaspBase::isTryError(fitTest)) NA else fitTest$ci.ub
+    ))
+  } else if (testType == "rankCorrelation") {
+    return(data.frame(
+      tau = if (jaspBase::isTryError(fitTest)) NA else fitTest$tau,
+      p   = if (jaspBase::isTryError(fitTest)) NA else fitTest$pval
+    ))
+  }
+}
+
+# compute power enhancement contours (lifted from zcurve)
+.power_to_z       <- function(power, alpha = .05, a = stats::qnorm(alpha/2,lower.tail = FALSE), two.sided = TRUE, nleqslv_control = list(xtol = 1e-15, maxit = 300, stepmax = .5)){
+  if(a  < 0)stop("a must be >= 0")
+  if(is.null(a) & is.null(alpha))stop("Either 'alpha' or 'a' must be provided")
+  if(is.null(alpha) & !is.null(a))alpha <- stats::pnorm(a, lower.tail = FALSE)*2
+  if(alpha < 0 | alpha > 1)stop("alpha must be >= 0 & <= 1")
+  if(!all(sapply(power, function(x)x >= alpha & x <= 1)))stop("power must be >= alpha & <= 1")
```

Comment:
Spaces

Reply:
I mean, these are functions defined in another package that I just copy pasted there, I would not bother renaming them and modifying them to JASP style. (I did not want to add the whole package into dependencies just for those three functions)

Reply:
I added the spaces to the rest of the code though